packages/block-editor/src/components/block-settings-menu-controls/edit-pattern-menu-item.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * WordPress dependencies
+ */
+import { MenuItem } from '@wordpress/components';
+import { _x } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import useContentOnlySectionEdit from '../../hooks/use-content-only-section-edit';
+
+export function EditPatternMenuItem( { clientId, onClose } ) {
+	const {
+		isSectionBlock,
+		isEditingContentOnlySection,
+		editContentOnlySection,
+	} = useContentOnlySectionEdit( clientId );
+
+	// Only show when the experiment is enabled, the block is a section block,
+	// and we're not already editing it
+	if (
+		! window?.__experimentalContentOnlyPatternInsertion ||
+		! isSectionBlock ||
+		isEditingContentOnlySection
+	) {
+		return null;
+	}
+
+	return (
+		<MenuItem
+			onClick={ () => {
+				editContentOnlySection( clientId );
+				onClose();
+			} }
+		>
+			{ _x( 'Edit pattern', 'Editing a pattern block in the Editor' ) }
```

Comment:
I like the updated copy.

@andrewserong raised an interesting [point](https://github.com/WordPress/gutenberg/pull/73183#issuecomment-3524115196) - are we referring to everything as patterns? E.g., template parts and `templateLock: 'contentOnly'`?

I'm fine with that by the way. 😄

Reply:
I'm thinking that for new users who may not know what a pattern is, it might be more clear to say "edit section", plus that includes patterns, template parts etc..